src/encoded/static/components/image.js

```diff
@@ -5,177 +5,149 @@ import * as globals from './globals';
+    lightboxVisible: PropTypes.bool,
+    /** Callback function to hide light box */
+    clearLightbox: PropTypes.func.isRequired,
+    /** URI of image to display in the light box */
+    lightboxImg: PropTypes.string.isRequired,
 };
 
 Lightbox.defaultProps = {
     lightboxVisible: false,
 };
 
 
-export class Attachment extends React.Component {
-    constructor() {
-        super();
-
-        // Set initial React state.
-        this.state = { lightboxVisible: false };
-
-        // Bind this to non-React methods.
-        this.lightboxClickImage = this.lightboxClickImage.bind(this);
-        this.clearLightbox = this.clearLightbox.bind(this);
-        this.handleEscKey = this.handleEscKey.bind(this);
-    }
-
-    // Register for keyup events for ESC key
-    componentDidMount() {
-        globals.bindEvent(window, 'keyup', this.handleEscKey);
-    }
-
-    // Unregister keyup events when component closes
-    componentWillUnmount() {
-        globals.unbindEvent(window, 'keyup', this.handleEscKey);
-    }
+export const Attachment = ({ context, attachment, className, showLink }) => {
+    const [lightboxVisible, setLightboxVisible] = React.useState(false);
 
     // Handle a click on the lightbox trigger (thumbnail)
-    lightboxClickImage(e) {
+    const lightboxClickImage = (e) => {
         e.preventDefault();
         e.stopPropagation();
-        this.setState({ lightboxVisible: true });
-    }
-
-    clearLightbox() {
-        this.setState({ lightboxVisible: false });
-    }
-
-    // If lightbox visible, ESC key closes it
-    handleEscKey(e) {
-        if (this.state.lightboxVisible && e.keyCode === 27) {
-            this.clearLightbox();
-        }
-    }
-
-    render() {
-        const { context, attachment, className, showLink } = this.props;
-        let attachmentHref;
-        let src;
-        let alt;
-        let height = '100';
-        let width = '100';
-        if (attachment && attachment.href && attachment.type) {
-            attachmentHref = url.resolve(context['@id'], attachment.href);
-            const attachmentType = attachment.type.split('/', 1)[0];
-            if (attachmentType === 'image' && attachment.type !== 'image/tiff') {
-                const imgClass = className ? `${className}-img` : '';
-                src = attachmentHref;
-                height = attachment.height || 100;
-                width = attachment.width || 100;
-                alt = 'Attachment from submitter';
-                if (!showLink) {
-                    // Just display the attachment image without any lightbox.
-                    return <img className={imgClass} src={src} height={height} width={width} alt={alt} />;
-                }
+        setLightboxVisible(true);
+    };
+
+    // Close the lightbox and image.
+    const clearLightbox = () => {
+        setLightboxVisible(false);
+    };
+
+    let attachmentHref;
+    let src;
+    let alt;
+    let height = '100';
```

Comment:
can some of these variables like "height" be in the if-statement?

Reply:
This is always the trade off. I rejiggered this function to use hooks, but I didn’t touch the guts of it to avoid having to specifically test that. But yes, you’re right. It probably was used outside the `if` at some point in the distant past.

Reply:
Acknowledged.